### PR TITLE
PLG-743: Populate Quality Score for product models on the first deployment

### PR DESCRIPTION
**Pull request by BryanS29-akeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Connector/Tasklet/RecomputeProductScoresTasklet.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ public function execute(): void
         $this->scheduleNextRecomputeProductsScoresJob($lastProductId);
     }
 
-    private function getNextProductIds($lastProductId): ProductIdCollection
+    private function getNextProductIds($lastProductId): array
```
Comment (antoinetrouve) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Connector/Tasklet/RecomputeProductScoresTasklet.php:80]: On peut typer le retour de la fonction de ce style @return array\<ProductIdCollection\>

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    private function CommandCanBeStarted(): bool
+    {
+        $query = <<<SQL
+SELECT 1 FROM pim_one_time_task WHERE code = :code
+SQL;
+
+        return !(bool)$this->dbConnection->executeQuery($query, ['code' => self::$defaultName])->fetchOne();
+    }
+
+    private function getNextProductModelIds(int $lastProductModelId): array
```
Comment (antoinetrouve) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php:107]: Peut-être typer le retour `array<int>` ?

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    protected function configure() :void
+    {
+        $this->setDescription('Populate scores for existing product models');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+
+        if (!$this->CommandCanBeStarted()) {
+            $output->writeln('This process has already been performed or is in progress.');
```
Comment (antoinetrouve) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php:39]: If the command is executed in background, i am not sure the output is necessary here.

**Review by antoinetrouve — APPROVED** [review]

**Comment by antoinetrouve** [comment]
There is no test for this command. IS it normal ?

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Connector/Tasklet/RecomputeProductScoresTasklet.php (hunk 1 above)
Comment (BryanS29-akeneo) [reply src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Connector/Tasklet/RecomputeProductScoresTasklet.php:80]: we return an array of int, i made a @return int[] in the phpdoc of the function

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php (hunk 3 above)
Comment (BryanS29-akeneo) [reply src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php:39]: we display it when we are in verbose

**Comment by BryanS29-akeneo** [comment]
> There is no test for this command. IS it normal ?

yes, we launch this command only one time and we already have a test on consolidate so it's not necessary

**Review by LaurentPetard — APPROVED** [review]

File: src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+SELECT 1 FROM pim_one_time_task WHERE code = :code
+SQL;
+
+        return !(bool)$this->dbConnection->executeQuery($query, ['code' => self::$defaultName])->fetchOne();
+    }
+
+    private function countPM(): int
+    {
+        $query = <<<SQL
+SELECT COUNT(*)
```
Comment (nicometriau) [inline src/Akeneo/Pim/Automation/DataQualityInsights/back/Infrastructure/Symfony/Command/PopulateProductModelScoresCommand.php:119]: You can select COUNT(1) which is lighter for the DB

**Review by nicometriau — APPROVED** [review]

**Pull request merged**